quiz/TypesAndUtils.scala

```diff
@@ -1,6 +1,14 @@
 package object genquiz {
 
-  type QuizName = String   // to understand these types see data in QuizData.scala
+  // to understand the types below, see data in QuizData.scala
+
+  case class QuizID(name: String, seed: Int) {
+    def this(name: String) = this(name, name.hashCode)
```

Comment:
Why an auxiliary constructor that overlaps with apply-factory?

Reply:
At first, I only had this auxiliary constructor, but then I added the apply-factory once I noticed that this auxiliary constructor didn't work for the "usual" way of creating instances of a case class (i.e. using `apply` instead of `new`). Is it preferable to just have the apply-factory? If so, I don't have anything against with removing this auxiliary constructor.

Reply:
I've removed the auxiliary constructor.